src/azure-cli-core/azure/cli/core/tests/test_keys.py

```diff
@@ -0,0 +1,135 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+import unittest
+import tempfile
+import paramiko
+import io
+import os
+import shutil
+from knack.util import CLIError
+
+from azure.cli.core.keys import generate_ssh_keys
+
+
+class TestGenerateSSHKeys(unittest.TestCase):
+
+    def setUp(self):
+        # set up temporary directory to be used for temp files.
+        self._tempdirName = tempfile.mkdtemp(prefix="key_tmp_")
+
+        self.key = paramiko.RSAKey.generate(2048)
+        keyOutput = io.StringIO()
+        self.key.write_private_key(keyOutput)
+
+        self.private_key = keyOutput.getvalue()
+        self.public_key = '{} {}'.format(self.key.get_name(), self.key.get_base64())
+
+    def tearDown(self):
+        # delete temporary directory to be used for temp files.
+        shutil.rmtree(self._tempdirName)
+
+    def test_public_key_file_exists(self):
```

Comment:
suggest name like test`_when_`public_key_file_exists? also this test can removed though as `generate_ssh_keys` should only be invoked when public key file doesn't exist

Reply:
I will update the test names.

I agree that the user should not call generate_ssh_keys even though they have a valid public key file. But I think it would still be good to handle and test this situation just in case this method is called by a future caller that fails to check this case. Again, I believe I was asked to handle all the different cases of public / private key being present or not present. ,